src/inputs/UploadFile/UploadFile.spec.js

```diff
@@ -24,18 +26,90 @@ const propsWithDirtyState = {
   ...defaultProps,
   isDirty: true,
 };
+
+const propsFileReadyToDownload = {
+  ...defaultProps,
+  file: {
+    name: 'file.csv',
+    status: UPLOAD_FILE_STATUS_KEY.READY_TO_DOWNLOAD,
+  },
+};
+
+const propsFileReadyToUpload = {
+  ...defaultProps,
+  file: {
+    name: 'file.csv',
+    status: UPLOAD_FILE_STATUS_KEY.READY_TO_UPLOAD,
+  },
+};
+
+const propsInvalidFileFormat = {
+  ...defaultProps,
+  error: {
+    message: 'Invalid file format',
+  },
+  file: {
+    name: 'invalid_file_format.png',
+    status: UPLOAD_FILE_STATUS_KEY.READY_TO_UPLOAD,
+  },
+};
+
+const propsFileActionProcessing = {
+  ...defaultProps,
+  file: {
+    name: 'file.csv',
+    status: UPLOAD_FILE_STATUS_KEY.DOWNLOADING,
+  },
+};
+
 const setUp = (props) => {
   renderInMuiThemeProvider(<UploadFile {...props} />);
 };
 
 describe('Checks file upload input in edit mode', () => {
-  test("Component is displayed in edit mode and dirtyInput class isn't applied when isDirty is false", () => {
+  test('dirtyInput class is applied when isDirty is true', () => {
+    setUp(propsWithDirtyState);
+    expect(uploadFileInputContainer.Container).toHaveDirtyInputClass();
+  });
+  test("Component is displayed in edit mode,dirtyInput class isn't applied when isDirty is false", () => {
     setUp(defaultProps);
+    expect(getByDataCy('download-button')).not.toBeInTheDocument();
+    expect(getByDataCy('delete-button')).not.toBeInTheDocument();
+    expect(getByDataCy('circular-progress')).not.toBeInTheDocument();
     expect(uploadFileInputContainer.Container).toBeInTheDocument();
     expect(uploadFileInputContainer.Container).not.toHaveDirtyInputClass();
+    expect(getByDataCy('file-error-message')).not.toBeInTheDocument();
   });
-  test('dirtyInput class is applied when isDirty is true', () => {
-    setUp(propsWithDirtyState);
-    expect(uploadFileInputContainer.Container).toHaveDirtyInputClass();
+  test('Component is correctly displayed when file is ready to upload', () => {
+    setUp(propsFileReadyToUpload);
+    expect(getByDataCy('file-name')).toBeInTheDocument();
+    expect(getByDataCy('delete-button')).toBeInTheDocument();
+    expect(getByDataCy('download-button')).not.toBeInTheDocument();
+    expect(getByDataCy('circular-progress')).not.toBeInTheDocument();
+    expect(getByDataCy('file-error-message')).not.toBeInTheDocument();
```

Comment:
I think we can use our Testing classes to get this elements : `TypographyTesting` for _file-name_ and _file-error-message_, `ButtonTesting` for _download-button_ and _delete-button_ :slightly_smiling_face: